lib/buildkite/test_collector/uploader.rb

```diff
@@ -54,5 +64,22 @@ def self.configure
     def self.tracer
       Thread.current[:_buildkite_tracer]
     end
+
+    def self.upload(data)
+      return false unless Buildkite::TestCollector.api_token
+
+      http = Buildkite::TestCollector::HTTPClient.new(Buildkite::TestCollector.url)
+
+      Thread.new do
+        response = begin
+          upload_attempts ||= 1
+          http.post_json(data)
+        rescue *Buildkite::TestCollector::Uploader::RETRYABLE_UPLOAD_ERRORS => e
+          if (upload_attempts += 1) > MAX_UPLOAD_ATTEMPTS
```

Comment:
Should this be `(upload_attempts += 1) <= MAX_UPLOAD_ATTEMPTS`?

Reply:
Ooo good catch!

Reply:
It's pretty hard to write a spec for this because its happening in a separate thread, any mock/stub/spy we can create in rspec won't be visible to this thread because its happening in a separate thread